clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/client/RegionScopedTitusClient.java

```diff
@@ -0,0 +1,202 @@
+import java.util.stream.Stream;
+
+
+public class RegionScopedTitusClient implements TitusClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(TitusRestAdapter.class);
+
+    /** Default connect timeout in milliseconds */
+    private static final long DEFAULT_CONNECT_TIMEOUT = 10000;
+
+    /** Default read timeout in milliseconds */
+    private static final long DEFAULT_READ_TIMEOUT = 20000;
+
+    /** An instance of {@link TitusRegion} that this RegionScopedTitusClient will use */
+    private final TitusRegion titusRegion;
+
+    /** Retrofit rest adapter for the titus environment */
+    private final TitusRestAdapter titusRestAdapter;
+
+    /** Actual connect timeout that will be used by the retrofit client */
+    private final long connectTimeoutMillis;
+
+    /** Actual read timeout that will be used by the retrofit client */
+    private final long readTimeoutMillis;
+
+    /** Titus client uses Jackson converter for retrofit. Titus client users can override this */
+    private final ObjectMapper objectMapper;
+
+    public RegionScopedTitusClient(TitusRegion titusRegion) {
+        this(titusRegion, DEFAULT_CONNECT_TIMEOUT, DEFAULT_READ_TIMEOUT, TitusClientObjectMapper.configure());
+    }
+
+    public RegionScopedTitusClient(TitusRegion titusRegion,
+                                   long connectTimeoutMillis,
+                                   long readTimeoutMillis,
+                                   ObjectMapper objectMapper) {
+
+        this.titusRegion = titusRegion;
+        this.connectTimeoutMillis = connectTimeoutMillis;
+        this.readTimeoutMillis = readTimeoutMillis;
+        this.objectMapper = objectMapper;
+        this.titusRestAdapter = createTitusRestAdapter(titusRegion);
+    }
+
+    private TitusRestAdapter createTitusRestAdapter(TitusRegion titusRegion) {
+        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor((msg) -> LOGGER.info("[{}] {}", titusRegion.getName(), msg));
+        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
+        OkHttpClient okHttpClient = new OkHttpClient.Builder()
+                .connectTimeout(connectTimeoutMillis, TimeUnit.MILLISECONDS)
+                .readTimeout(readTimeoutMillis, TimeUnit.MILLISECONDS)
+                .addNetworkInterceptor(interceptor)
+                .build();
+        return new Retrofit.Builder()
+                .baseUrl(titusRegion.getEndpoint())
+                .client(okHttpClient)
+                .addConverterFactory(JacksonConverterFactory.create(objectMapper))
+                .validateEagerly(true)
+                .build()
+                .create(TitusRestAdapter.class);
+    }
+
+
+    // APIs
+    // ------------------------------------------------------------------------------------------
+
+    @Override
+    public Job getJob(String jobId) {
+        return execute(titusRestAdapter.getJob(jobId));
+    }
+
+    private <T> T execute(Call<T> call) {
```

Comment:
This is my dirty first pass to provide a more retrofit1 like experience on retrofit2 (e.g. give me the unwrapped call result or throw an exception)